geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/records/SimpleFeatureRecordConverter.scala

```diff
@@ -87,21 +157,32 @@ object SimpleFeatureRecordConverter {
         values.put(converters(i).field.getFieldName, converters(i).toRecord(feature.getAttribute(i)))
         i += 1
       }
-      new MapRecord(schema, values, false, false)
+      new SimpleFeatureMapRecord(feature, schema, values)
     }
 
     override def convert(feature: Record): SimpleFeature = {
       val raw = feature.getValues
       val values = Array.ofDim[AnyRef](converters.length)
       var i = 0
       while (i < converters.length) {
-        values(i) = converters(i).toAttribute(raw(i + 1))
+        // Why is this raw(i+1)?
+        values(i) = converters(i).toAttribute(raw(i))
```

Comment:
i think this still needs to be +1 to skip over the fid, no?

Reply:
In the unit test I wrote, it was breaking with it as +1....

Reply:
maybe the by name case is different than the by attribute case? seems odd that the unit test hasn't changed though... can you dig into the old unit test and figure out what's going on?